doc/migrations/v0.26-v0.27.md

```diff
@@ -0,0 +1,182 @@
+
+**Before**
+```js
+libp2p.start((err) => {
+  if (err) throw err
+  console.log('libp2p started')
+})
+```
+
+**After**
+```js
+await libp2p.start()
+console.log('libp2p started')
+```
+
+## Pull Streams to Streaming Iterables
+
+The libp2p API no longer supports Pull Streams and has migrated to [Streaming Iterables][streaming_iterable]. If you would like to continue using Pull Streams in your application code, or need additional time to migrate your code base, you can leverage the conversion modules [async-iterator-to-pull-stream](https://github.com/alanshaw/async-iterator-to-pull-stream) and [pull-stream-to-async-iterator](https://github.com/alanshaw/pull-stream-to-async-iterator).
+
+For a growing list of async iterator modules, you should follow the [it-awesome repo][it_awesome].
+
+## Sample API Migrations
+
+### Registering Protocol Handlers
+
+Protocol registration is very similar to how it previously was, however, the handler now takes a single parameter containing the incoming stream and its protocol. Additionally, you can now pass an array of protocols to `.handle`, but a single string is still supported.
+
+**Before**
+```js
+const pull = require('pull-stream')
+libp2p.handle('/echo/1.0.0', (protocol, conn) => pull(conn, conn))
+```
+
+**After**
+```js
+const pipe = require('it-pipe')
+libp2p.handle(['/echo/1.0.0'], ({ protocol, stream }) => pipe(stream, stream))
+```
+
+### Dialing and Sending Data
+
+`dialProtocol` no longer takes a callback, and will now return a [Streaming Iterable][streaming_iterable] and the protocol that was successfully negotiated. The new stream can be used with async iterator modules, see [it-awesome][it_awesome], instead of pull streams.
+
+**Before**
+```js
+const pull = require('pull-stream')
+libp2p.dialProtocol(peerInfo, '/echo/1.0.0', (err, conn) => {
+  if (err) { throw err }
+
+  console.log('nodeA dialed to nodeB on protocol: /echo/1.0.0')
+
+  pull(
+    pull.values(['hey']),
+    conn,
+    pull.collect((err, data) => {
+      if (err) { throw err }
+      console.log('received echo:', data.toString())
+    })
+  )
+})
+```
+
+**After**
+```js
+const pipe = require('it-pipe')
+const { protocol, stream } = await libp2p.dialProtocol(peerInfo, '/echo/1.0.0')
+await pipe(
+  ['hey'],
+  stream,
+  async function (source) {
+    for await (const data of source) {
+      console.log('received echo:', data.toString())
+    }
+  }
+)
+```
+
+### Checking if a peer is connected
+
+`peerInfo.isConnected` is unreliable. It requires an address be set via `PeerInfo.connect(addr)`, which requires the PeerInfo to be already tracking that address. This is problematic for incoming connections. The remoteAddr of an incoming connection is not reliable, and shouldn't be stored on the PeerInfo. Instead, we should simply check for an actual open connection. Since libp2p has better tracking of connections, this is trivial to do.
```

Comment:
I would maybe rephrase this around `peerInfo.isConnected` being deprecated and that consumers should not use it because libp2p is now tracking this information centrally and it will not be correct.